charts/portworx/templates/storage-cluster.yaml

```diff
@@ -1,193 +1,84 @@
+  imagePullSecret: {{ .Values.global.image.pullSecret }}
   kvdb:
-  {{- if eq $internalKVDB true }}
-    internal: true
-  {{- else }}
-    internal: false
-    {{- if empty $etcdEndPoints }}
-    "{{ required "A valid ETCD url in the format etcd:http://<your-etcd-endpoint> is required. Verify that the key is correct and there isnt any typo in specifying that, also ensure it is accessible from all node of your kubernetes cluster" .etcdEndPoint}}"
-    {{- else }}
+    internal: {{ .Values.kvdb.internal }}
+    {{- with .Values.etcd.endpoints }}
     endpoints:
-      {{- $endpoints := $etcdEndPoints | split ";" }}
-      {{- range $key, $val := $endpoints }}
-      - {{$val}}
-      {{- end }}
+      {{- toYaml . | nindent 6 }}
     {{- end }}
-    {{- if ne $kvdbAuthSecret "none" }}
     authSecret: {{ .Values.kvdb.authSecretName }}
-    {{- end }}
-  {{- end }}
-
-{{- if and (ne $drives "none") (not (hasPrefix "/" $drives)) }}
-  cloudStorage:
-    deviceSpecs:
-      {{- $driveNames := $drives | split ";" }}
-      {{- range $index, $name := $driveNames }}
-      - {{ $name }}
-      {{- end }}
-    {{- if and (ne $kvdbDevice "none") (not (hasPrefix "/" $kvdbDevice)) }}
-    kvdbDeviceSpec: {{ $kvdbDevice }}
-    {{- end }}
-    {{- if and (ne $journalDevice "none") (not (hasPrefix "/" $journalDevice)) }}
-    journalDeviceSpec: {{ $journalDevice }}
-    {{- end }}
-    {{- if ne (.Values.maxStorageNodesPerZone | int) 0 }}
-    maxStorageNodesPerZone: {{ .Values.maxStorageNodesPerZone }}
-    {{- end }}
-{{- else }}
-  {{- if $AKSInstall }}
+  {{- with .Values.cloudStorage }}
   cloudStorage:
-    deviceSpecs:
-      - type=Premium_LRS,size=150
-    {{- if and (ne $kvdbDevice "none") (not (hasPrefix "/" $kvdbDevice)) }}
-    kvdbDeviceSpec: {{ $kvdbDevice }}
-    {{- end }}
-    {{- if and (ne $journalDevice "none") (not (hasPrefix "/" $journalDevice)) }}
-    journalDeviceSpec: {{ $journalDevice }}
-    {{- end }}
-    {{- if gt (.Values.maxStorageNodesPerZone | int) 0 }}
-    maxStorageNodesPerZone: {{ .Values.maxStorageNodesPerZone }}
-    {{- end }}
-  {{- else }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with .Values.storage }}
   storage:
-    {{- if hasPrefix "/" $drives }}
-    devices:
-      {{- $driveNames := $drives | split ";" }}
-        {{- range $index, $name := $driveNames }}
-        - {{ $name }}
-        {{- end }}
-    {{- end }}
-    {{- if or $usefileSystemDrive $deployEnvironmentIKS }}
-    forceUseDisks: true
-    {{- end }}
-    {{- if eq $usedrivesAndPartitions true }}
-    useAllWithPartitions: true
-    {{- else }}
-    useAll: true
-    {{- end }}
-    {{- if and (ne $kvdbDevice "none") (hasPrefix "/" $kvdbDevice) }}
-    kvdbDevice: {{ $kvdbDevice }}
-    {{- end }}
-    {{- if and (ne $journalDevice "none") (or (hasPrefix "/" $journalDevice) (eq "auto" $journalDevice)) }}
-    journalDevice: {{ $journalDevice }}
-    {{- end }}
+    {{- toYaml . | nindent 4 }}
```

Comment:
Did we test this? Following cases should be covered:

- Specifying disks as /dev/sda
- With type=gp2,size=150,tag=key
- not specifying any drive spec

Reply:
I only have a limited set of cluster setups to test this on. Is the portworx team able to test this against the various types of supported clusters? AKS, OpenShift, etc.